### PR TITLE
Add comments to customer class

### PR DESCRIPTION
**Pull request by jf-viguier**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add comments to customer class
| Type?             | refacto
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | N/A
| How to test?      | N/A
| Possible impacts? | None, just comments
| Sponsor company   | Creabilis.com

### CODE REVIEW

File: classes/Customer.php [hunk 1]
```diff
@@ -610,6 +610,9 @@ public static function customerHasAddress($idCustomer, $idAddress)
         return self::$_customerHasAddress[$key];
     }
 
+    /**
+     * Reset static cache
```
Comment (matks) [inline classes/Customer.php]: Is this comment really useful as the function name is `resetStaticCache` ? 😄

File: classes/Customer.php [hunk 2]
```diff
@@ -435,7 +435,7 @@ public static function getCustomers($onlyActive = null)
      *
      * @param string $email e-mail
      * @param string $plaintextPassword Password is also checked if specified
-     * @param bool $ignoreGuest
+     * @param bool $ignoreGuest To exclude guest customer
```
Comment (FabienPapet) [inline classes/Customer.php]:
```suggestion
     * @param bool $ignoreGuest to ignore guest customers
```

File: classes/Customer.php [hunk 3]
```diff
@@ -565,7 +565,7 @@ public static function isBanned($idCustomer)
      * Check if e-mail is already registered in database.
      *
      * @param string $email e-mail
-     * @param bool $returnId
+     * @param bool $returnId Return Customer ID if true, or boolean
```
Comment (FabienPapet) [inline classes/Customer.php]: Misplaced comment

File: classes/Customer.php [hunk 4]
```diff
@@ -1130,7 +1138,7 @@ public static function getDefaultGroupId($idCustomer)
     }
 
     /**
-     * Get current Country.
+     * Get current Country or default country
```
Comment (FabienPapet) [inline classes/Customer.php]:
```suggestion
     * Get current country or default country
```

File: classes/Customer.php (hunk 1 above)
Comment (jf-viguier) [reply classes/Customer.php]: @matks it's not a good practice to comment every method for phpdoc automatic doc generation ?

File: classes/Customer.php (hunk 4 above)
Comment (jf-viguier) [reply classes/Customer.php]: done

File: classes/Customer.php (hunk 3 above)
Comment (jf-viguier) [reply classes/Customer.php]: @FabienPapet no, this parameter ask to return an int or a boolean. I've changed to a clearer explanation

File: classes/Customer.php (hunk 2 above)
Comment (jf-viguier) [reply classes/Customer.php]: done

File: classes/Customer.php (hunk 1 above)
Comment (jf-viguier) [reply classes/Customer.php]: comment removed @matks

**Review by nicosomb — APPROVED** [review]

**Comment by jf-viguier** [comment]
The PHP / PHPUnit (8.1) (pull_request) fail doesn't seems related to my pr, can someone launch CI ?

**Comment by Progi1984** [comment]
@jf-viguier Done.

**Comment by Progi1984** [comment]
@jf-viguier The CI is now fixed

@PrestaShop/prestashop-maintainers Open to review #hacktoberfest

**Review by FabienPapet — APPROVED** [review]

**Review by FabienPapet — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Pull request merged**